src/coreclr/gc/gc.cpp

```diff
@@ -44422,7 +44422,27 @@ void gc_heap::verify_heap (BOOL begin_gc_p)
             uint8_t*        curr_object = heap_segment_mem (seg);
             uint8_t*        prev_object = 0;
 
+            bool verify_bricks_p = true;
 #ifdef USE_REGIONS
+            assert (!heap_segment_read_only_p (seg));
```

Comment:
let's make this `FATAL_GC_ERROR` too.
```suggestion
            if (heap_segment_read_only_p (seg))
            {
                dprintf (1, ("seg %Ix is ro! Shouldn't happen with regions", (size_t)seg));
                FATAL_GC_ERROR();
            }
```

Reply:
ugh, no matter how many times I tried github refuses to apply this diff: 
![image](https://user-images.githubusercontent.com/523221/194053967-114cd151-17da-46f2-be26-ca06756a2b18.png)

so I'm going to apply by hands and add you as co-authored